### DISCUSSION TITLE
Is there a global event listener for invalidateQueries in TanStack Query?

### DISCUSSION THREAD

**Discussion by ezalorsara** [TanStack/query#7798] [Q&A] [answered]
Hi TanStack Query team,

I'm currently using TanStack Query in my project and I have a question regarding the invalidateQueries function. Specifically, I would like to know if there is a global event listener or hook that triggers when the invalidateQueries function is fired.

Context:
I want to perform some side effects whenever any query is invalidated throughout my application. Having a global event listener would make this process more manageable and reduce the need to manually handle this logic in multiple places.

Question:
Is there an existing global event listener or hook within TanStack Query that triggers when invalidateQueries is called?
If not, what would be the recommended approach to implement such functionality?
Thank you for your assistance!

**Comment by TkDodo** [comment, marked as answer]
no, there is not. I'd make a wrapper around `queryClient.invalidateQuries` that also does whatever you want to do. But this imperative approach is not something that we have internally. For a query, it doesn't matter what triggered it. An invalidation also doesn't necessarily mean that the queryFn is running

**Reply by ezalorsara** [reply]
Thanks Man, appreciate your reply
